### PR TITLE
feat: test for index.json generation

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #221 
Added more tests for index.json generation and made several fixes (I've missed the highest level json object with `count` and `items` at first). The example of this json can be checked [here](https://learn.microsoft.com/en-us/nuget/api/registration-base-url-resource#sample-response) in docs.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@dgarus please, review

File: src/main/java/com/artipie/nuget/IndexJson.java [hunk 1]
```diff
@@ -137,8 +138,10 @@ private static List<JsonObject> sortedPackages(final JsonObject newest,
             final String version, final JsonObject old) {
             final JsonArray arr;
```
Comment (dgarus) [inline src/main/java/com/artipie/nuget/IndexJson.java:139]: Suggest rewrite this method

```
private static List<JsonObject> sortedPackages(final JsonObject newest,
            final String version, final JsonObject old) {
            List<JsonObject> list = Collections.singletonList(newest);
            if (old.containsKey(Update.ITEMS)) {
                final JsonArray items = old.getJsonArray(Update.ITEMS);
                if (!items.isEmpty()) {
                    final JsonObject value = items.get(0).asJsonObject();
                    final JsonArray arr;
                    if (value.containsKey(Update.ITEMS)
                        && !(arr = value.getJsonArray(Update.ITEMS)).isEmpty()) {
                        list = new ArrayList<>(arr.size() + 1);
                        arr.stream().map(JsonValue::asJsonObject)
                            .filter(val -> !version.equals(version(val))).forEach(list::add);
                        list.add(newest);
                        list.sort(Comparator.comparing(val -> new Semver(version(val))));
                    }
                }
            }
            return list;
        }
```

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
Minor comment.

**Comment by olenagerasimova** [comment]
@dgarus thanks, can we agree on two `if's`? Three conditions seems too much...

**Comment by dgarus** [comment]
> @dgarus thanks, can we agree on two `if's`? Three conditions seems too much...

ok

**Pull request merged**